evidence/errors.go

```diff
@@ -0,0 +1,24 @@
+package evidence
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/types"
+)
+
+type ErrInvalidEvidence struct {
```

Comment:
why expose?

Reply:
Okay, perhaps I need a better understanding of when errors should be exposed or not because almost all the errors in all other packages are exposed, so I have just been copying them

Reply:
I think in general it's just good practice. Unless, however, you know with 100% certainty, the error will never be used/needed by the outside (package) world.

Reply:
I think you only want to expose things when YOU know they will be used / needed. otherwise, keep them private

Reply:
If we'll be creating a typed error every time we return an error in public function, we will end up with pile of extra code, which is not needed in reality